src/gui/qgsfieldformwidget.h

```diff
@@ -0,0 +1,256 @@
+/***************************************************************************
+   qgsfieldformwidget.h
+    --------------------------------------
+    begin                : September 2020
+    copyright            : (C) 2020 by Ivan Ivanov
+    email                : ivan@opengis.ch
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************/
+
+#ifndef QGSFIELDFORMWIDGET_H
+#define QGSFIELDFORMWIDGET_H
+
+#include "ui_qgsfieldformwidget.h"
+#include "qgsguiutils.h"
+#include "qgsapplication.h"
+
+#include "qgis_gui.h"
+#include "qgsfield.h"
+
+#include <QRegExpValidator>
+#include <QRegExp>
+
+class QgsVectorLayer;
+
+/**
+* \ingroup gui
+* Dialog to set up parameters to create a new field in a vector layer, and on accept() to create it and add it to the layers
+* \note unstable API (will likely change)
+* \since QGIS 3.16
+*/
+class GUI_EXPORT QgsFieldFormWidget : public QWidget, private Ui::QgsFieldFormWidgetBase
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * List of advanced fields to be shown. If `Neither` is chosen, the whole advanced fields UI is hidden.
+     */
+    enum class AdvancedField
+    {
+      // sip complains if it is called `None`
+      Neither = 1 << 0,
```

Comment:
Do we actually need this? You could use AdvancedFields() with no flags as a way to indicate they should all be hidden

Reply:
Good to know, thanks